test/test_torch.py

```diff
@@ -4600,6 +4570,36 @@ def test_masked_select_discontiguous(self, device):
                 torch.masked_select(v, m, out=out_dc)
                 self.assertEqual(out_dc, expected, atol=0, rtol=0)
 
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_masked_fill(self, device, dtype):
+        dt = dtype
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            for dtype in [torch.uint8, torch.bool]:
+                num_dest = 10
+                dst = torch.tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=dt)
+                mask = torch.rand(num_dest).mul(2).floor().to(dtype)
+                val = random.random()
+                dst2 = dst.clone()
+
+                dst.masked_fill_(mask, val)
+                for i in range(num_dest):
+                    if mask[i]:
+                        dst2[i] = val
+                self.assertEqual(dst, dst2, atol=0, rtol=0)
+
+                # test non-contiguous case
+                dst = torch.randn(num_dest, num_dest, num_dest).permute((2, 0, 1))
+                dst2 = dst.clone()
```

Comment:
since clone by default will preserve layout, this test is not testing much - it's calling similar operations on similar tensors